lib/ts/translation/translationContext.tsx

```diff
@@ -29,29 +29,29 @@ export const TranslationContextProvider: React.FC<
     }>
 > = ({ children, defaultLanguage, userTranslationFunc, defaultStore, translationControlEventSource }) => {
     const [translationStore, setTranslationStore] = useState<TranslationStore>(defaultStore);
-    const cookieLang = getCurrentLanguageFromCookie();
-    const [currentLanguage, setCurrentLanguage] = useState(cookieLang === null ? defaultLanguage : cookieLang);
+    const [currentLanguage, setCurrentLanguage] = useState<string | undefined>(undefined);
 
-    const translateFunc = useCallback<TranslationFunc>(
-        (key: string) => {
-            if (userTranslationFunc !== undefined) {
-                return userTranslationFunc(key);
-            }
-            const res = translationStore[currentLanguage] && translationStore[currentLanguage][key];
-            const fallback = translationStore[defaultLanguage] && translationStore[defaultLanguage][key];
-
-            if (res === undefined) {
-                if (fallback !== undefined) {
-                    return fallback;
-                }
-                return key;
-            }
+    const loadLanguageFromCookies = useCallback(
```

Comment:
No need for `useCallback`, this can be just defined inside the `useEffect`, which could safely use the same dependency array, since this doesn't overwrite anything if the current language is set.

Reply:
Done. We only need this to run on mount so not using any dependencies

Reply:
I think it's better to add the dependency array if it doesn't cause problems. Otherwise, we should start to disable the lint rule specifically, since I think it'd be great to turn it back on.
```
// eslint-disable-next-line react-hooks/exhaustive-deps
```
It's not really an issue, you can resolve this if you like.

Reply:
Fair enough, added the dependencies on defaultLanguage and setCurrentLanguage. Like you said, it doesnt affect anything so doesnt hurt